compositor_render/src/transformations/shader/validation.rs

```diff
@@ -51,6 +55,20 @@ fn validate_globals(
             .map_err(ShaderValidationError::GlobalBadType)?;
     }
 
+    // validate user-defined buffer is a uniform
+    shader
+        .global_variables
+        .iter()
+        .find(|(_, global)| {
+            global.binding.is_some()
+                && global.binding.as_ref().unwrap().group == USER_DEFINED_BINDING.0
+                && global.binding.as_ref().unwrap().binding == USER_DEFINED_BINDING.1
+        })
+        .map_or(Ok(()), |(_, global)| match global.space {
+            naga::AddressSpace::Uniform => Ok(()),
+            _ => Err(ShaderValidationError::UserBindingNotUniform),
+        })?;
```

Comment:
your solution is reversed, we would need to check for space == Uniform, not !=. Even after the fix, this would report an error when the user doesn't define a binding, which we don't want.

Reply:
I think changing it will make it even more complex, so im leaving it as-is